plans/action/install.pp

```diff
@@ -141,15 +141,15 @@
   # puppet and are present in PuppetDB, it is not necessary anymore.
   $puppetdb_database_temp_config = {
     'puppet_enterprise::profile::database::puppetdb_hosts' => (
-      $compiler_targets + $primary_target + $primary_replica_target
+      $compiler_targets + $primary_target + $replica_target
     ).map |$t| { $t.peadm::certname() },
   }
 
   $primary_pe_conf = peadm::generate_pe_conf({
     'console_admin_password'                                          => $console_password,
     'puppet_enterprise::puppet_master_host'                           => $primary_target.peadm::certname(),
     'pe_install::puppet_master_dnsaltnames'                           => $dns_alt_names,
-    'puppet_enterprise::puppetdb_database_host'                       => $puppetdb_database_target.peadm::certname(),
+    'puppet_enterprise::primary_postgresql_host'                      => $primary_postgresql_target.peadm::certname(),
```

Comment:
I think this is the problem. An erroneous substitution.

While things "mostly" work with this changed, it causes the configuration on the primary to oscillate which PostgreSQL server is backing PuppetDB between runs, in effect making data "shimmer". This shimmer means the primary's orchestrator configuration is out of sync when the replica provision task is run, which forces the orchestrator to restart during the provision. That action disconnects the `puppet infra provision` command mid-task, causing it to throw an error.

Ironically, everything is still more or less working correctly after that error is thrown. Eventual consistency for 'ya.

```suggestion
    'puppet_enterprise::puppetdb_database_host'                       => $primary_postgresql_target.peadm::certname(),
```